pkg/shared/utils/arrays/generic_array_utils.go

```diff
@@ -1,12 +1,86 @@
 package arrays
```

Comment:
Maybe we can rename this package to `slices`?

Reply:
I will rename it in another PR since that will require changing all the references.